pinot-hadoop-filesystem/src/main/java/org/apache/pinot/filesystem/HadoopPinotFS.java

```diff
@@ -92,9 +92,25 @@ public boolean delete(URI segmentUri, boolean forceDelete)
   @Override
   public boolean move(URI srcUri, URI dstUri, boolean overwrite)
       throws IOException {
-    if (exists(dstUri) && !overwrite) {
+    if (!exists(srcUri)) {
```

Comment:
This part of the code exists in all implementations. Can we add this to PinotFS, call super.move() and then handle FS specific logic in the impl classes?

Reply:
Common logic has been moved to base class.